File: Client/MessageService.cs
using KolibSoft.Catalogue.Client;
using KolibSoft.MsgHub.Core;

namespace KolibSoft.MsgHub.Client;

public class MessageService : CatalogueService<MessageModel, MessageFilters>
{

    public MessageService(HttpClient httpClient, string uri) : base(httpClient, uri) { }

}

File: docs/lib/message_service.js
import { MessageModel } from "./message_model.js";
import { CatalogueService } from "./modules/catalogue.js";

class MessageService extends CatalogueService {

    /**
     * @param {{(input: RequestInfo | URL, init?: RequestInit | undefined) => Promise<Response>}} fetch 
     * @param {string} uri 
     */
    constructor(fetch, uri) {
        super(json => new MessageModel(json), fetch, uri);
    }

}

export {
    MessageService
}